spec.md

```diff
@@ -0,0 +1,73 @@
+# Collaborative clock and transport proof-of-concept
+
+This document lays out requirements for a proof-of-concept clock and transport for web audio applications.
+
+## Goals
+
+The goal is to create a simple application where two users collaborate on a shared data structure that syncs over a time-based loop. 
+
+## Definitions
+
+- Server - a backend application that stores and syncs the data structure
+- Client - a frontend application that plays and updates its part of the data structure
+- Shared Data - the shared data structure
+- Local Data - the local version of the data structure. Each client has their own version of Local Data.
+- Clock - a thing that keeps time. Each client uses one to play over the data structure.
+- Loop Start - the time when the loop is started or restarted after loop end
+
+## Constraints
+
+- Local Data and Shared Data use the same schema
+- Clients modify independent parts of the schema. The data structure is shared, but to simplify synchronization, each user can only modify one part of the schema.
```

Comment:
What does it mean for the data structure to be shared?

Reply:
Was having trouble writing this clearly. The idea is that clients "own" one part of the data and can only modify that part. Will push an update like:

```
Clients own a section of the schema. A Client may only modify data that corresponds to a 
section they own. The goal is to simplify synchronization and prevent data tug-of-wars.
```